internal/pkg/template/diff/override.go

```diff
@@ -40,58 +43,185 @@ func (m *ignorer) parse(_, _ *yaml.Node, _ string, _ overrider) (diffNode, error
+type getAttConverter struct {
+	intrinsicFuncMapTagConverter
+}
+
+// match returns true if both from node and to node are calling the "GetAtt" intrinsic function.
+// "GetAtt" only accepts either sequence or scalar, therefore match returns false if either of from and to has invalid 
+// input node to "GetAtt".
+// Example1: "!GetAtt" and "!GetAtt" returns true.
+// Example2: "!GetAtt" and "Fn::GetAtt" returns true.
+// Example3: "!Ref" and "!GetAtt" returns false.
+// Example4: "!GetAtt [a,b]" and "Fn::GetAtt:a:b" returns false because the input type is wrong.
+func (converter *getAttConverter) match(from, to *yaml.Node, key string, overrider overrider) bool {
+	if !converter.intrinsicFunc.match(from, to, key, overrider) {
+		return false
+	}
+	if intrinsicFuncName(from) != "GetAtt" {
+		return false
+	}
+	fromValue, toValue := from, to
+	if from.Kind == yaml.MappingNode {
+		fromValue = from.Content[1] // A valid full-form intrinsic function always contain a child node.
+	}
+	if to.Kind == yaml.MappingNode {
+		toValue = to.Content[1]
+	}
+	switch {
+	case fromValue.Kind == yaml.ScalarNode && toValue.Kind == yaml.ScalarNode:
+		fallthrough
+	case fromValue.Kind == yaml.SequenceNode && toValue.Kind == yaml.SequenceNode:
+		fallthrough
+	case fromValue.Kind == yaml.SequenceNode && toValue.Kind == yaml.ScalarNode:
+		fallthrough
+	case fromValue.Kind == yaml.ScalarNode && toValue.Kind == yaml.SequenceNode:
+		return true
+	default:
+		return false
+	}
+}
+
+// parse compares two nodes that call the "GetAtt" function. Both from and to can be written in either full or short form.
+// parse assumes that from and to are already matched by getAttConverter.
+func (converter *getAttConverter) parse(from, to *yaml.Node, key string, overrider overrider) (diffNode, error) {
+	// Extract the input node to GetAtt.
+	fromValue, toValue := from, to
+	if from.Kind == yaml.MappingNode {
+		fromValue = from.Content[1] // A valid full-form intrinsic function always contain a child node.
+	}
+	if to.Kind == yaml.MappingNode {
+		toValue = to.Content[1]
+	}
+	// If the input node are of the same type (i.e. both seq or both scalar), parse them normally.
+	if fromValue.Kind == toValue.Kind {
+		return converter.intrinsicFuncMapTagConverter.parse(from, to, key, overrider)
+	}
+	// Otherwise, first convert the scalar input to seq input, then parse.
+	var err error
+	switch {
+	case fromValue.Kind == yaml.ScalarNode:
+		fromValue, err = getAttScalarToSeq(fromValue)
+	case toValue.Kind == yaml.ScalarNode:
+		toValue, err = getAttScalarToSeq(toValue)
 	}
 	if err != nil {
 		return nil, err
 	}
+	diff, err := parse(fromValue, toValue, intrinsicFunctions["GetAtt"].fullName, overrider)
 	if diff == nil {
-		return nil, nil
+		return nil, err
 	}
 	return &keyNode{
 		keyValue:   key,
 		childNodes: []diffNode{diff},
 	}, nil
 }
 
+func intrinsicFuncName(node *yaml.Node) string {
```

Comment:
I feel like it would be safer for us to model thing like "!Join" and "Fn::Join" as different types (2 types initiated by calling regex matching and error out earlier if one doesn't match) and call methods to get func name or convert into the other type 💭

Reply:
That is an option but that'd be like 20+ overriders to process for each YAML nodes :(, so I didn't opt for this route.

> it would be safer

I agree that the the logics that handle each intrinsic functions will be more decoupled this way, but why is it necessarily safer? 

> error out earlier if one doesn't match

Can you elaborate on this?

Reply:
i think i'm partial to just the `TrimPrefix`, but i wonder if instead of a function that trims _both_ "Fn::" and "!" we should explicitly trim the correct value - if it's in the tag form, won't it always mean we should strip "!"? And "Fn::" in the long form?

Reply:
ah I think I get it now. I understood this comment wrong!

With the current implementation, the `override` would  unexpectedly `match` a snippet like this:
```yaml
Sub: 123
```

In this example, `Sub` is either written incorrectly, or is just the name of a field. We should only match `Fn::Sub` for a full form. 

Good catch!

I will combine @iamhopaul123 and @dannyrandall 's suggestion 👍🏼